parser/parser.go

```diff
@@ -245,6 +249,64 @@ func parseUnary(p *Parser) (ast.Node, error) {
 	return ast.NewUnaryExpr(tok.Type, expr), nil
 }
 
+func parseVarDecls(p *Parser) (ast.Node, error) {
+	p.forget(1)
+	return _parseVarDecls(p)
+}
+
+func _parseVarDecls(p *Parser) (ast.VarDecls, error) {
+
+	if !p.scry(2) {
+		return nil, fmt.Errorf("parser var decl: expected at least two tokens got[%s]", p.lookahead)
+	}
+
+	identifier := p.lookahead[0]
+	possibleAssignment := p.lookahead[1]
+	p.forget(2)
+
+	if identifier.Type != token.Ident {
+		return nil, fmt.Errorf("parser: var decl: expected identifier got[%s]", identifier)
+	}
+
+	varname := ast.NewIdent(identifier.Value)
+	if possibleAssignment.Type == token.SemiColon {
+		return ast.NewVarDecls(ast.NewVarDecl(varname, ast.NewUndefined())), nil
+	}
+
+	if possibleAssignment.Type != token.Assign {
+		return nil, fmt.Errorf("parser: var decl: expected assignment token [=] got [%s]", possibleAssignment)
+	}
+
+	p.scry(1)
+	assignExpr := p.lookahead[0]
+	parser, hasparser := varAssignParsers[assignExpr.Type]
+
+	if !hasparser {
+		return nil, fmt.Errorf("parser: var decl: invalid token[%s] expected assigment expression", assignExpr)
+	}
+
+	val, err := parser(p)
+	if err != nil {
+		return nil, fmt.Errorf("parser: var decl: error[%s] parsing variable assign expression", err)
+	}
+
+	res := ast.NewVarDecls(ast.NewVarDecl(varname, val))
+	p.scry(1)
+	possibleSemiColon := p.lookahead[0]
+	p.forget(1)
+
+	if possibleSemiColon.Type == token.SemiColon || possibleSemiColon.Type == token.EOF {
+		return res, nil
+	}
+
+	if possibleSemiColon.Type != token.Comma {
+		return nil, fmt.Errorf("parser: var decl: invalid token[%s] expected comma", possibleSemiColon)
+	}
+
+	vars, err := _parseVarDecls(p)
```

Comment:
I wanted to be able to do this recursion, it does not expect a "var" token and it can't return an ast.Node (not exactly can't but it don't seem like a good idea to me). The name is kinda lame but I was unable to think on something better until now. I thought about varAssigment since it parsers "a = <val>" not expecting the "var", but I imagine that the kind of node produced must be different since there is a difference on Javascript between declaring multiple variables starting with "var" and without (reference errors if the variable does not exist and it is not initialized).

Reply:
http://es5.github.io/#x12.2

Spec call this section of declaration as `VariableDeclarationList`. What about `parseDeclList` ?